src/v6_indexer/backfillStakingRecords.ts

```diff
@@ -0,0 +1,264 @@
+import { schema, db, eq, and, or, inArray } from "@metadaoproject/indexer-db";
+import { futarchyClient, connection } from "./connection";
+import { FUTARCHY_PROGRAM_ID } from "@metadaoproject/futarchy/v0.6";
+import { PublicKey, ConfirmedSignatureInfo } from "@solana/web3.js";
+import { log } from "../logger/logger";
+import { index } from "./indexer";
+import pLimit from "p-limit";
+
+const logger = log.child({
+  module: "v6_backfillStakingRecords"
+});
+
+const limit = pLimit(2); // Rate limiting for RPC calls
+const delay = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+/**
+ * Gap fill historical stake transactions by processing all stake-related signatures
+ */
+export async function gapFillStakeTransactions(): Promise<{message:string, error: Error | undefined}> {
+  logger.info(`Gap filling all historical stake transactions`);
+  
+  try {
+    // Get all stake accounts from chain to identify which proposals have stakes
+    const stakeAccounts = await futarchyClient.autocrat.account.stakeAccount.all();
+    logger.info(`Found ${stakeAccounts.length} stake accounts on chain`);
+    
+    // Extract unique proposal addresses that have stakes
+    const proposalAddresses = [...new Set(stakeAccounts.map(s => s.account.proposal.toBase58()))];
+    logger.info(`Found stakes across ${proposalAddresses.length} unique proposals`);
+    
+    let totalSignaturesProcessed = 0;
+    let totalEventsProcessed = 0;
+    
+    // For each proposal, fetch all historical signatures
+    for (const proposalAddr of proposalAddresses) {
+      logger.info(`Processing historical transactions for proposal ${proposalAddr}`);
+      
+      const proposalPubkey = new PublicKey(proposalAddr);
+      let signatures: ConfirmedSignatureInfo[] = [];
+      let lastSignature: string | undefined;
+      
+      // Fetch all signatures for this proposal
+      while (true) {
+        const batch = await connection.getSignaturesForAddress(
+          proposalPubkey,
+          { 
+            limit: 1000,
+            before: lastSignature
+          },
+          "finalized"
+        );
+        
+        if (batch.length === 0) break;
+        
+        signatures = signatures.concat(batch);
+        lastSignature = batch[batch.length - 1].signature;
+        
+        logger.info(`Fetched ${signatures.length} signatures so far for proposal ${proposalAddr}`);
+      }
+      
+      logger.info(`Found ${signatures.length} total signatures for proposal ${proposalAddr}`);
+      
```

Comment:
Okay here's the change I want you to make. We want to fetch signatures on the program_id from the signature_accounts table and then use that as the comparison before we just start slamming signatures. We want to know what we have before we start going through this. before we make the call to index. (basically avoiding already indexed data and saturating the RPC with getTransaction